ftplugin/forth.vim

```diff
@@ -16,19 +16,47 @@ setlocal commentstring=\\\ %s
 setlocal comments=s:(,mb:\ ,e:),b:\\
 setlocal iskeyword=33-126,128-255
 
-let b:undo_ftplugin = "setl cms< com< isk<"
+let s:include_patterns =<< trim EOL
+
+  \<\%(INCLUDE\|REQUIRE\)\>\s\+\zs\k\+\ze
+  \<S"\s\+\zs[^"]*\ze"\s\+\%(INCLUDED\|REQUIRED\)\>
+EOL
+let &l:include = $'\c{ s:include_patterns[1:]->join('\|') }'
+
+let s:define_patterns =<< trim EOL
+  :
+  [2F]\=CONSTANT
+  [2F]\=VALUE
+  [2F]\=VARIABLE
+  BEGIN-STRUCTURE
+  BUFFER:
+  CODE
+  CREATE
+  MARKER
+  SYNONYM
+EOL
+let &l:define = $'\c\<\%({ s:define_patterns->join('\|') }\)'
+
+" assume consistent intra-project file extensions
+let &l:suffixesadd = "." .. expand("%:e")
+
+let b:undo_ftplugin = "setl cms< com< def< inc< isk< sua<"
 
 if exists("loaded_matchit") && !exists("b:match_words")
+  let s:matchit_patterns =<< trim EOL
+
+    \<\:\%(NONAME\)\=\>:\<EXIT\>:\<;\>
+    \<IF\>:\<ELSE\>:\<THEN\>
+    \[IF]:\[ELSE]:\[THEN]
+    \<?\=DO\>:\<LEAVE\>:\<+\=LOOP\>
+    \<CASE\>:\<ENDCASE\>
+    \<OF\>:\<ENDOF\>
+    \<BEGIN\>:\<WHILE\>:\<\%(AGAIN\|REPEAT\|UNTIL\)\>
+    \<CODE\>:\<END-CODE\>
```

Comment:
in forth-standard.org, it's `CODE` and `;CODE`

Reply:
Unless I'm overlooking something `;CODE` is used like `DOES>` but for assembly code.

I did notice some time ago that DurexForth and a few others appeared to be using it for their implementation-defined `CODE` ending and was going to investigate further before adding it as a middle match to `:`.

Reply:
Ah! You are of course right, I got it wrong.

END-CODE may originally be from Ragsdale's assembler as described here: http://www.forth.org/fd/FD-V03N5.pdf